domain/consensus/processes/transactionvalidator/mass.go

```diff
@@ -20,29 +18,30 @@ func (v *transactionValidator) transactionMassStandalonePart(tx *externalapi.Dom
 	return size*v.massPerTxByte + totalScriptPubKeySize*v.massPerScriptPubKeyByte
 }
 
-func (v *transactionValidator) transactionMass(tx *externalapi.DomainTransaction) (uint64, error) {
+func (v *transactionValidator) transactionMass(tx *externalapi.DomainTransaction) uint64 {
 	if transactionhelper.IsCoinBase(tx) {
-		return 0, nil
+		return 0
 	}
 
-	standaloneMass := v.transactionMassStandalonePart(tx)
-	sigOpsCount := uint64(0)
-	var missingOutpoints []*externalapi.DomainOutpoint
-	for _, input := range tx.Inputs {
-		utxoEntry := input.UTXOEntry
-		if utxoEntry == nil {
-			missingOutpoints = append(missingOutpoints, &input.PreviousOutpoint)
-			continue
-		}
-		// Count the precise number of signature operations in the
-		// referenced public key script.
-		sigScript := input.SignatureScript
-		isP2SH := txscript.IsPayToScriptHash(utxoEntry.ScriptPublicKey())
-		sigOpsCount += uint64(txscript.GetPreciseSigOpCount(sigScript, utxoEntry.ScriptPublicKey(), isP2SH))
+	// calculate mass for size
+	size := estimatedsize.TransactionEstimatedSerializedSize(tx)
+	massForSize := size * v.massPerTxByte
+
+	// calculate mass for scriptPubKey
+	totalScriptPubKeySize := uint64(0)
+	for _, output := range tx.Outputs {
+		totalScriptPubKeySize += 2 //output.ScriptPublicKey.Version (uint16)
```

Comment:
I know this is not related to this PR, but do we really want to give extra weight to the version number?

Reply:
As long as it is kept in the UTXOSet (And I believe it is), then absolutely.

The block mass reflects the social cost, and the fact something is mandatory or not doesn't change the social cost.

Reply:
The question is why do we give extra weight script pub key size. If it's related to script execution, script version has nothing to do with it. And if it's related to UTXO set size, we should also give the same weight to the output value field.

Reply:
The answer is it's all about UTXOSet size.
Script execution limiting is through SigOpCount (the rest of the Ops are negligable).

So I've created an issue:
https://github.com/kaspanet/kaspad/issues/1796